src/test/java/de/bonndan/nivio/input/RelationResolverTest.java

```diff
@@ -0,0 +1,108 @@
+import org.mockito.Mockito;
+import org.slf4j.Logger;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static de.bonndan.nivio.model.ServiceItems.pick;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+class RelationResolverTest {
+
+    private RelationResolver relationResolver;
+
+    @Mock
+    ProcessLog log;
+
+    @BeforeEach
+    public void setup() {
+        log = new ProcessLog(Mockito.mock(Logger.class));
+        relationResolver = new RelationResolver(log);
+    }
+
+    @Test
+    public void assignTemplateWithRegex() {
+
+        LandscapeDescription landscapeDescription = getLandscapeDescriptionWithAppliedTemplates("/src/test/resources/example/example_templates2.yml");
+        relationResolver.processRelations(landscapeDescription);
+
+        ItemDescription one = (ItemDescription) pick("crappy_dockername-78345", null, landscapeDescription.getItemDescriptions());
+        assertNotNull(one);
+        assertEquals("alpha", one.getGroup());
+
+        ItemDescription two = (ItemDescription) pick("crappy_dockername-2343a", null, landscapeDescription.getItemDescriptions());
+        assertNotNull(two);
+        assertEquals("alpha", two.getGroup());
+
+        ItemDescription three = (ItemDescription) pick("other_crappy_name-2343a", null, landscapeDescription.getItemDescriptions());
+        assertNotNull(three);
+        assertEquals("beta", three.getGroup());
+    }
+
+
+
+
+    @Test
+    public void resolvesTemplatePlaceholdersInProviders() {
+
+        LandscapeDescription landscapeDescription = getLandscapeDescriptionWithAppliedTemplates("/src/test/resources/example/example_templates2.yml");
+        relationResolver.processRelations(landscapeDescription);
+
+        //the provider has been resolved using a query instead of naming a service
+        ItemDescription providedbyBar = (ItemDescription) pick("crappy_dockername-78345", null, landscapeDescription.getItemDescriptions());
+        assertNotNull(providedbyBar);
+        Set<RelationItem<String>> relations = providedbyBar.getRelations(RelationType.PROVIDER);
+        assertNotNull(relations);
+        assertEquals(1, relations.size());
+        RelationItem s = relations.iterator().next();
+        assertEquals("nivio:templates2/alpha/crappy_dockername-2343a", s.getSource());
+    }
+
+    @Test
+    public void resolvesTemplatePlaceholdersInDataflow() {
+
+        LandscapeDescription landscapeDescription = getLandscapeDescriptionWithAppliedTemplates("/src/test/resources/example/example_templates2.yml");
+        relationResolver.processRelations(landscapeDescription);
+
+        ItemDescription hasdataFlow = (ItemDescription) pick("crappy_dockername-78345", null, landscapeDescription.getItemDescriptions());
```

Comment:
At the risk of appearing fussy: Correct camel case: `hasDataFlow`. 😁

Reply:
Such things can be fixed by anybody without PR in the develop branch later.